### PR TITLE
Split the installation section & add the bootstrap sub section

### PR DESCRIPTION
**Pull request by mehak151**
Made the following structural changes as per: https://github.com/fluxcd/website/issues/1560

Installation
- index (Prerequisites, CLI install, install options: explain flux bootstrap vs terraform flux_bootstrap_git vs flux install)
- Bootstrap
-- index (empty, will just list all options below)
-- Generic Git Server
-- GitHub
-- GitLab
-- BitBucket
-- AWS CodeCommit
-- Azure DevOps
-- Google Cloud Source
- Upgrade (CLI bootstrap, GH Action, Renovate, terraform, in-cluster)
- Uninstall (CLI uninstall)

### CODE REVIEW

File: content/en/flux/installation/_index.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
+---
+title: "Installation"
+description: "Flux install, bootstrap, upgrade and uninstall documentation."
+weight: 30
+---
+
+This guide walks you through setting up Flux to
+manage one or more Kubernetes clusters.
```
Comment (stefanprodan) [inline content/en/flux/installation/_index.md:8]: This is no longer a guide style doc. Can you please write here an introduction and explain what Flux is made of, and give a quick overview of what people should expect from this doc.

File: content/en/flux/installation/_index.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+[`zsh`](./cmd/flux_completion_zsh.md), [`fish`](./cmd/flux_completion_fish.md),
+and [`powershell`](./cmd/flux_completion_powershell.md)
+are also supported with their own sub-commands.
+
+A container image with `kubectl` and `flux` is available on DockerHub and GitHub:
+
+* `docker.io/fluxcd/flux-cli:<version>`
+* `ghcr.io/fluxcd/flux-cli:<version>`
+
+## Bootstrap
```
Comment (stefanprodan) [inline content/en/flux/installation/_index.md:96]:
```suggestion
## Bootstrap with Flux CLI
```

File: content/en/flux/installation/_index.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+architectures.
+
+If your Git provider is **AWS CodeCommit**, **Azure DevOps**, **Bitbucket Server**, **GitHub** or **GitLab** please
+follow the specific bootstrap procedure:
+
+* [AWS CodeCommit](./bootstrap/aws-code-commit.md#flux-installation-for-aws-codecommit)
+* [Azure DevOps](./bootstrap/azure.md#flux-installation-for-azure-devops)
+* [Bitbucket Server and Data Center](./bootstrap/bitbucket.md#bitbucket-server-and-data-center)
+* [GitHub.com and GitHub Enterprise](./bootstrap/github.md#github-and-github-enterprise)
+* [GitLab.com and GitLab Enterprise](./bootstrap/gitlab.md#gitlab-and-gitlab-enterprise)
```
Comment (stefanprodan) [inline content/en/flux/installation/_index.md:118]: We need to list here all the options with Generic Git server first, also no need to repeat the provider names in bold.

File: content/en/flux/installation/_index.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+* `docker.io/fluxcd/flux-cli:<version>`
+* `ghcr.io/fluxcd/flux-cli:<version>`
+
+## Bootstrap
+
+Using the `flux bootstrap` command you can install Flux on a
+Kubernetes cluster and configure it to manage itself from a Git
+repository.
+
```
Comment (stefanprodan) [inline content/en/flux/installation/_index.md:101]: Here we need a short example, you could use the bootstrap git with a private SSH key so we avoid talking about SSH agent here.

File: content/en/flux/installation/_index.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+  path                   = "clusters/my-cluster"
+  network_policy         = true
+  kustomization_override = file("${path.module}/kustomization.yaml")
+}
+```
+
+For more details on how to use the Terraform provider
+please see the [Flux docs on registry.terraform.io](https://registry.terraform.io/providers/fluxcd/flux/latest/docs).
+
+## Customize Flux manifests
```
Comment (stefanprodan) [inline content/en/flux/installation/_index.md:154]: This should be in the "Configuration" page.

File: content/en/flux/installation/_index.md [hunk 6]
```diff
@@ -0,0 +1,308 @@
...
+  --path=clusters/my-cluster
+```
+
+To make further amendments, pull the changes locally,
+edit the `kustomization.yaml` file, push the changes upstream
+and rerun bootstrap or let Flux upgrade itself.
+
+Checkout the [bootstrap cheatsheet](../cheatsheets/bootstrap) for various examples of how to customize Flux.
+
+### Multi-tenancy lockdown
```
Comment (stefanprodan) [inline content/en/flux/installation/_index.md:208]: This should be in the "Configuration/Multi-tenancy lockdown" page.

File: content/en/flux/installation/bootstrap/_index.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+---
+title: "Bootstrap"
+linkTitle: "Bootstrap"
+description: ""
```
Comment (stefanprodan) [inline content/en/flux/installation/bootstrap/_index.md:4]: Please add a description here.

File: content/en/flux/installation/bootstrap/aws-code-commit.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
+---
+title: AWS Code Commit
+linkTitle: AWS Code Commit
+description: "How to follow bootstrap procedure for Flux using AWS Code Commit"
```
Comment (stefanprodan) [inline content/en/flux/installation/bootstrap/aws-code-commit.md:4]:
```suggestion
description: "How to bootstrap Flux with AWS Code Commit"
```

Please change all the others pages to this shorter format.

File: content/en/flux/installation/bootstrap/generic-git-server.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
+---
+title: Generic Git Server
+linkTitle: Generic Git server
+description: "Generic Git Server"
```
Comment (stefanprodan) [inline content/en/flux/installation/bootstrap/generic-git-server.md:4]:
```suggestion
description: ""How to bootstrap Flux with a generic Git server"
```

File: content/en/flux/installation/bootstrap/github.md [hunk 10]
```diff
@@ -0,0 +1,78 @@
+---
+title: GitHub
+linkTitle: GitHub
+description: "How to follow bootstrap procedure for Flux using GitHub"
+weight: 20
+---
+### GitHub and GitHub Enterprise
```
Comment (stefanprodan) [inline content/en/flux/installation/bootstrap/github.md:7]:
```suggestion

```

File: content/en/flux/installation/bootstrap/github.md [hunk 11]
```diff
@@ -0,0 +1,78 @@
+---
+title: GitHub
```
Comment (stefanprodan) [inline content/en/flux/installation/bootstrap/github.md:2]:
```suggestion
title: GitHub and GitHub Enterprise
```

File: content/en/flux/installation/bootstrap/github.md (hunk 10 above)
Comment (stefanprodan) [reply content/en/flux/installation/bootstrap/github.md:7]: Make the same change on all guides please, we don't want to duplicate the title.

File: content/en/flux/installation/upgrade.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
+---
+title: Upgrade
+linkTitle: Upgrade (CLI bootstrap, GH Action, Renovate, terraform, in-cluster)
+description: "Upgrade (CLI bootstrap, GH Action, Renovate, terraform, in-cluster)"
```
Comment (stefanprodan) [inline content/en/flux/installation/upgrade.md:4]:
```suggestion
linkTitle: Upgrade
description: "Upgrade Flux using bootstrap"
```

File: content/en/flux/installation/_index.md (hunk 5 above)
Comment (mehak151) [reply content/en/flux/installation/_index.md:154]: Noted. You will see it on the Configuration page when I create a PR for it.

File: content/en/flux/installation/_index.md (hunk 6 above)
Comment (mehak151) [reply content/en/flux/installation/_index.md:208]: Noted. You will see it on the configuration page when I create a PR for it.

**Comment by stefanprodan** [comment]
@mehak151 please fix Error: error building site: assemble: /opt/build/repo/content/en/flux/installation/bootstrap/generic-git-server.md:2:1: failed to unmarshal YAML: yaml: line 2: did not find expected key

**Comment by stefanprodan** [comment]
@mehak151 please see #1573, this one can be closed

**Pull request closed without merging**